### ISSUE TITLE
admission: integrate TTL scans with AC read bandwidth limiter

### ISSUE THREAD

**Issue by aadityasondhi** [cockroachdb/cockroach#121782]
We have seen full table scans that occur as part of row-level TTL cause overload by saturating the read bandwidth on a node.

**Comment by rafiss** [comment]
Just to clarify, TTL does not do a full table scan, per se. Each leaseholder node of the table scans only the ranges it holds the lease for. There are two levels of concurrency: 
- Work is spread across leaseholders, which all do their scans at the same time.
- Within each leaseholder, there are `GOMAXPROCS` goroutines working concurrently.

This parallelism is probably one of the reasons why read bandwidth can get saturated.

**Comment by rafiss** [comment]
Also, each of the goroutines mentioned above will delete the keys from the range immediately after it scans the range. Does the TTL code also need to integrate with the disk write traffic limiter?

**Comment by rafiss** [comment]
Is this issue still relevant now that we're going with https://github.com/cockroachdb/cockroach/pull/167853?

**Comment by wenyihu6** [comment]
Yes, +1 on closing. We have decided to go with https://github.com/cockroachdb/cockroach/pull/167853. Read more on the decision in https://docs.google.com/document/d/11UanJEWezTyu6nv-M5G52SIQRCKiBR2LJ9VQHbFjp-4/edit?tab=t.0.
